lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -277,6 +289,91 @@ def __get_k8s_job_spec_template(self, ajs):
         k8s_spec_template["metadata"]["annotations"].update(extra_metadata.get('annotations', {}))
         return k8s_spec_template
 
+    def __get_k8s_service_spec(self, ajs):
+        """The k8s spec template is nothing but a Service spec, except that it is nested and does not have an apiversion
+        nor kind."""
+        if not ajs.job_wrapper.guest_ports:
```

Comment:
```suggestion
        guest_ports = ajs.job_wrapper.guest_ports
        if not guest_ports:
```
I had missed that you're using them on line 313